### PR TITLE
Fix store invite

### PR DESCRIPTION
**Pull request by h1ppox99**

### CODE REVIEW

File: packages/matrix-identity-server/src/db/index.ts [hunk 1]
```diff
@@ -507,6 +510,53 @@ class IdentityServerDb<T extends string = never>
...
+          id,
+          address,
+          data: JSON.stringify(data)
+        })
+        .then(() => {
+          resolve(id)
+        })
+        .catch((err) => {
+          /* istanbul ignore next */
+          this.logger.error('Failed to insert token', err)
```
Comment (guimard) [inline packages/matrix-identity-server/src/db/index.ts:532]: Missing "reject": a promise must always call resolve or reject

File: packages/matrix-identity-server/src/db/index.ts [hunk 2]
```diff
@@ -507,6 +510,53 @@ class IdentityServerDb<T extends string = never>
...
+        .then((rows) => {
+          /* istanbul ignore else */
+          if (rows.length > 0) {
+            resolve(JSON.parse(rows[0].data as string))
+          } else {
+            reject(new Error('Unknown token'))
+          }
+        })
+        .catch((e) => {
+          reject(e)
```
Comment (guimard) [inline packages/matrix-identity-server/src/db/index.ts:555]: Add logger.error ?

File: packages/matrix-identity-server/src/ephemeral_signing/index.ts [hunk 3]
```diff
@@ -41,37 +41,31 @@ const SignEd25519 = <T extends string = never>(
...
-                }
+                )
               })
               .catch((err) => {
                 /* istanbul ignore next */
-                idServer.logger.error(
-                  'Error while fetching one-time token',
-                  err
-                )
+                idServer.logger.error('Token denied', err)
```
Comment (guimard) [inline packages/matrix-identity-server/src/ephemeral_signing/index.ts:66]: WHy ignore line here, unable to reproduce in test ?

File: packages/matrix-identity-server/src/index.test.ts [hunk 4]
```diff
@@ -1250,6 +1264,38 @@ describe('Use configuration file', () => {
...
+        })
+        // @ts-expect-error mock is unknown
+        fetch.mockImplementation(async () => await mockResponse)
+        await mockResponse
+        const response = await request(app)
+          .post('/_matrix/identity/v2/store-invite')
+          .set('Authorization', `Bearer ${validToken}`)
+          .set('Accept', 'application/json')
+          .send({
+            address: '+33612345678',
```
Comment (guimard) [inline packages/matrix-identity-server/src/index.test.ts:1286]: "address" looks a bad name here

File: packages/matrix-identity-server/src/invitation/index.ts [hunk 5]
```diff
@@ -78,48 +94,71 @@ const mailBody = (
       .replace(/__date__/g, new Date().toUTCString())
       // initialize message id
       .replace(/__messageid__/g, randomString(32))
-      .replace(/__room_name__/g, room_name ?? '')
-      .replace(/__room_avatar__/g, room_avatar ?? '')
-      .replace(/__room_type__/g, room_type ?? '')
+      .replace(/__room_name__/g, room_name)
+      .replace(/__room_avatar__/g, room_avatar)
+      .replace(/__room_type__/g, room_type)
+      .replace(/__link__/g, inviteLink('matrix.to', sender_user_id, room_alias))
```
Comment (guimard) [inline packages/matrix-identity-server/src/invitation/index.ts:100]: `matrix.to` must be a config parameter, not a fixed string

File: packages/matrix-identity-server/src/invitation/index.ts [hunk 6]
```diff
@@ -134,6 +173,7 @@ const StoreInvit = <T extends string = never>(
   return (req, res) => {
     idServer.authenticate(req, res, (_data, _id) => {
       jsonContent(req, res, idServer.logger, (obj) => {
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
         validateParameters(res, schema, obj, idServer.logger, async (obj) => {
           const _address = (obj as storeInvitationArgs).address
```
Comment (guimard) [inline packages/matrix-identity-server/src/invitation/index.ts:178]: I think we shouldn't use the same fields name `address` for a phone number

**Comment by guimard** [comment]
Please add a link to your MR in related issues when exist _(#62)_ here

File: packages/matrix-identity-server/src/ephemeral_signing/index.ts (hunk 3 above)
Comment (h1ppox99) [reply packages/matrix-identity-server/src/ephemeral_signing/index.ts:66]: this was already tested actually - I removed the Istanbul ignore

File: packages/matrix-identity-server/src/index.test.ts (hunk 4 above)
Comment (h1ppox99) [reply packages/matrix-identity-server/src/index.test.ts:1286]: The name has to be the same for the different media. But the spec only supports mail so this is named address for the moment.

File: packages/matrix-identity-server/src/invitation/index.ts (hunk 6 above)
Comment (h1ppox99) [reply packages/matrix-identity-server/src/invitation/index.ts:178]: Should we create a different endpoint for phone maybe ?

File: packages/matrix-identity-server/src/index.test.ts (hunk 4 above)
Comment (guimard) [reply packages/matrix-identity-server/src/index.test.ts:1286]: Looking at others API, `phone` is used to designate phone when type is "pĥone"

Comment (h1ppox99) [reply packages/matrix-identity-server/src/index.test.ts:1286]: I think the most common use is "country" (The two-letter uppercase ISO-3166-1 alpha-2 country code that the number in phone_number should be parsed as if it were dialled from) + "phone_number"

File: packages/matrix-identity-server/src/invitation/index.ts (hunk 6 above)
Comment (guimard) [reply packages/matrix-identity-server/src/invitation/index.ts:178]: Can also be a different schema. If type is email, address is required, else phone is required

File: packages/matrix-identity-server/src/index.test.ts (hunk 4 above)
Comment (guimard) [reply packages/matrix-identity-server/src/index.test.ts:1286]: No, there is an encoding convention for phone in Matrix spec: international number without `+` or spaces

File: packages/matrix-identity-server/src/invitation/index.ts (hunk 6 above)
Comment (h1ppox99) [reply packages/matrix-identity-server/src/invitation/index.ts:178]: But then the token I store will have two fields (phone and address) and one of them will be empty. Is it ok ?

Comment (guimard) [reply packages/matrix-identity-server/src/invitation/index.ts:178]: Are we talking about database schema or API field names ?

Comment (h1ppox99) [reply packages/matrix-identity-server/src/invitation/index.ts:178]: I am now talking about how I store the invitation in the db, should I do the same thing as for the API field names, ie add a new column phone which will sometimes be empty, or keep one column for addresses which will store email as well as phone numbers ?

Comment (guimard) [reply packages/matrix-identity-server/src/invitation/index.ts:178]: I was talking about API fieldnames. For DB, do what is useful for you

Comment (h1ppox99) [reply packages/matrix-identity-server/src/invitation/index.ts:178]: In the synapses db, the convention is to use a medium and an address, address being either an email address or a phone number. I will stick to this usage

**Review by guimard — APPROVED** [review]

**Comment by guimard** [comment]
Remember to drop "draft" mode for review

**Pull request merged**